web-service/helpers.py

```diff
@@ -54,13 +54,10 @@ def append_invalid():
             append_invalid()
             continue
 
-        siteInfo = models.SiteInformation.get_by_id(url)
-
-        if force or siteInfo is None:
-            siteInfo = FetchAndStoreUrl(siteInfo, url)
+        siteInfo = GetSiteInfoForUrl(url, rssi, txpower, force_update)
 
         if siteInfo is None:
-            append_invalid()
+            #append_invalid()
```

Comment:
This change is significant.

It means that 404's, 204, etc, are now *filtered out* of the PWS response.  I think this should be the behaviour, but it may affect the experience with e.g. local urls that cannot be hit from PWS but can be from the client on an intranet.

This problem could also be solved locally on the client, perhaps, by detecting local-only urls somehow?